File: app/Controller/DescriptionsController.php
<?php
class DescriptionsController extends AppController {

  public $name = 'Descriptions';

  function beforeFilter() {
    if ($this->params['teacher']) {
      $this->isTeacherFilter();
      if ($this->action == 'teacher_create') {
        $this->isOwningSubjectFilter();
      }
    }
  }

  function isOwningSubjectFilter()
  {
    $this->loadModel('Subject');
    if (!$this->Subject->findByIdAndTeacherId($this->request->data['Description']['subject_id'], $this->currentUser('id'))) {
      $this->Session->setFlash('Brak dostępu.', 'flash_error');
      $this->redirect($this->referer());
    }
  }

  function teacher_create() {
    if ($this->request->is('post')) {
  $this->request->data['Description']['semester_id'] = $this->currentUser('semester_id');
      $this->Description->create();
      if ($this->Description->save($this->request->data)) {
        $this->Session->setFlash('Dodano ocenę.', 'flash_success');
        $this->redirect($this->referer());
      }
    }
  }

  function teacher_delete() {
    if ($this->Description->delete($this->params['pass'][0])) {
      $this->redirect($this->referer());
    } else {
      $this->redirect($this->referer());
    }
  }

}

File: app/controllers/teacher/descriptions_controller.rb
class Teacher::DescriptionsController < ApplicationController

  before_filter :authenticate_teacher!
  before_filter :teacher_has_active_class?, :school_class_has_active_semester?, :school_class_has_students?

  def create
    subject = current_teacher.subjects.find(params[:description][:subject_id])
    @description = subject.descriptions.build(params[:description])
    @description.save
    redirect_to :back
  end

  def destroy
    description = current_teacher.school_class.semester.descriptions.find(params[:id])
    subject = description.subject
    redirect_to teacher_subject_path(subject) if description.destroy
  end

end
